Trace `.tsx` input files

tylerc opened the pull request:
This fixes #39 the issue I just filed around `.tsx` files being unable to be used as the entrypoint for a route.

I noticed in `@now/node` this callback was responsible for compiling my `.ts` files, but it was never getting called for my `.tsx` files: https://github.com/zeit/now-builders/blob/canary/packages/now-node/src/index.ts#L177

This pull request makes `node-file-trace` aware that `.tsx` is a valid file extension for TypeScript files.

tylerc commented:
Pinging @styfle here as well.

styfle commented on src/node-file-trace.js:
```diff
@@ -27,7 +27,7 @@ module.exports = async function (files, opts = {}) {
     const path = resolve(file);
     job.emitFile(job.realpath(path), 'initial');
-    if (path.endsWith('.js') || path.endsWith('.node') || job.ts && path.endsWith('.ts'))
+    if (path.endsWith('.js') || path.endsWith('.node') || job.ts && (path.endsWith('.ts') || path.endsWith('.tsx')))
```
Thanks for finding and fixing this issue! Can you add a unit test for this?

tylerc replied:
Sure! Give me a bit to figure out how your tests are organized and I'll push something up.

tylerc commented:
@styfle unit tests have been pushed. Since it didn't quite fit the mold of the other tests, I had to add another call to `it()` in `unit.test.js`, hopefully that's okay.

I also found a couple other places that referenced `.ts` but not `.tsx`, so I fixed those up as well.

I couldn't figure out how to test the one in `analyze.js` (nothing ever seemed to get to that code) but both changes in `node-file-trace.js` are covered inthe unit test I added.

styfle commented on test/unit/tsx2/output.js:
```diff
@@ -0,0 +1,4 @@
+[
+    "test\\unit\\tsx2\\dep.tsx",
```
```suggestion
    "test/unit/tsx2/dep.tsx",
```

styfle commented on test/unit/tsx2/output.js:
```diff
@@ -0,0 +1,4 @@
+[
+    "test\\unit\\tsx2\\dep.tsx",
+    "test\\unit\\tsx2\\input.tsx"
```
```suggestion
    "test/unit/tsx2/input.tsx"
```

styfle commented on test/unit/tsx2/input.tsx:
```diff
@@ -0,0 +1,3 @@
+import * as fs from "fs";
```
Let's rename this directory from `tsx2` to `tsx-input`.

tylerc replied:
Will do!

styfle commented on test/unit.test.js:
```diff
@@ -32,6 +32,48 @@ for (const unitTest of fs.readdirSync(join(__dirname, 'unit'))) {
     console.log('skipping symlink test on Windows: ' + unitTest);
     continue;
   }
+  if (unitTest === 'tsx2') {
```
As far as I can tell, the only difference between this unit test and the others is that there is a `readFileMock`.

So instead of copy/paste, let's keep it [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and instead add a `expect(readFileMock.mock.calls.length).toBe(2);` when the unitTest is the tsx one.

Does that make sense?

tylerc replied:
Yes, it does. I'll make that change and push it up.

tylerc commented:
@styfle All your suggestions should now be integrated.

tylerc commented:
Thank you!
